roles/nova-data/tasks/libvirt.yml

```diff
@@ -56,29 +58,157 @@
+  file:
+    path: "{{ item }}"
+    state: directory
+    owner: root
+    group: kvm
+  with_items:
+    - /etc/pki/CA
+    - /etc/pki/libvirt/private
+
+- name: get uname -n
+  command: uname -n
+  register: uname
+
+- name: create 'uname -n' as fact 'uname_host'
+  set_fact:
+    uname_host: "{{ uname.stdout }}"
+
+- name: install CA file
+  template:
+    src: cacert.pem.j2
+    dest: /etc/pki/CA/cacert.pem
+    owner: root
+    group: kvm
+    mode: 0644
+  register: ca_install
+
+- name: upload CA key
+  template:
+    src: cacert.key.j2
+    dest: /tmp/cacert.key
+  register: ca_key_install
+
+- name: upload server certificate template
+  template:
+    src: server-cert-template.j2
+    dest: /tmp/server-cert-template
+  register: ca_template
+
+- name: create server cert private key
+  shell: "certtool --generate-privkey > /etc/pki/libvirt/private/serverkey.pem"
+  when: ca_install|changed or ca_key_install|changed or ca_template|changed
+
+- name: create server cert
+  command: "certtool --generate-certificate
+           --template /tmp/server-cert-template
+           --load-privkey /etc/pki/libvirt/private/serverkey.pem
+           --load-ca-certificate /etc/pki/CA/cacert.pem
+           --load-ca-privkey /tmp/cacert.key
+           --outfile /etc/pki/libvirt/servercert.pem"
+  when: ca_install|changed or ca_key_install|changed or ca_template|changed
+  notify: restart libvirt-bin
+
+- name: upload client certificate template
+  template:
+    src: client-cert-template.j2
+    dest: /tmp/client-cert-template
+  register: client_template
+
+- name: create client private key
+  shell: "certtool --generate-privkey > /etc/pki/libvirt/private/clientkey.pem"
+  when: client_template|changed
+
+- name: create client cert
+  command: "certtool --generate-certificate
+           --template /tmp/client-cert-template
+           --load-privkey /etc/pki/libvirt/private/clientkey.pem
+           --load-ca-certificate /etc/pki/CA/cacert.pem
+           --load-ca-privkey /tmp/cacert.key
+           --outfile /etc/pki/libvirt/clientcert.pem"
+  when: client_template|changed
+  notify: restart libvirt-bin
+
+- name: set dir owner
+  file:
+    dest: /etc/pki/libvirt
+    recurse: yes
+    owner: root
+    group: kvm
+
+- name: set perms
+  file:
+    dest: "{{ item.value.path }}"
+    mode: "{{ item.value.mode }}"
+  with_dict:
+    1:
+      path: /etc/pki/libvirt
+      mode: "0755"
+    2:
+      path: /etc/pki/libvirt/private
+      mode: "0750"
+    3:
+      path: /etc/pki/libvirt/servercert.pem
+      mode: "0440"
+    4:
+      path: /etc/pki/libvirt/private/serverkey.pem
+      mode: "0440"
+    5:
+      path: /etc/pki/libvirt/clientcert.pem
+      mode: "0400"
+    6:
+      path: /etc/pki/libvirt/private/clientkey.pem
+      mode: "0400"
+
+- name: remove temporary files
+  file:
+    path: "{{ item }}"
+    state: absent
+  with_items:
+    - /tmp/cacert.key
+    - /tmp/server-cert-template
+    - /tmp/client-cert-template
```

Comment:
Not sure we'd want to leave them there either.  At the very least the key should be deleted.